app/controllers/application_controller/filter.rb

```diff
@@ -947,7 +947,11 @@ def adv_search_build(model)
     # Restore @edit hash if it's saved in @settings
     @expkey = :expression                                               # Reset to use default expression key
     if session[:adv_search] && session[:adv_search][model.to_s]
-      @edit = copy_hash(session[:edit])
```

Comment:
@hstastna To continue to look for the best solution, one would need a sound knowledge of all the use cases that you could possibly hit.
IMO, it's hard to predict all use cases beforehand and come up with a catch-all solution. 
Plus, this being a `session` based code makes it even more dicey :(

The best approach would be to refactor it and make is `session`-less. But that would take a lot of time, as it would need studying the code thoroughly.

That said, the best we can do at this point, is rely on some historical data. 
Meaning, if the line `@edit = copy_hash(session[:adv_search][model.to_s])` was not changed for 3 years or so, then I would say, it is somewhat safe to assume that, that was stable code. 

This PR puts that change back in with a reasonable conditional around it in order to retain your fix in #776. So let's go ahead with that.

I'll address the rubocop. Thanks.